src/Application/ApplicationFileProcessor.php

```diff
@@ -118,6 +120,10 @@ public function processFiles(
         ?callable $preFileCallback = null,
         ?callable $postFileCallback = null
     ): ProcessResult {
+        if (SimpleParameterProvider::provideArrayParameter(Option::FILE_PATHS) === []) {
+            SimpleParameterProvider::setParameter(Option::FILE_PATHS, $filePaths);
+        }
```

Comment:
On parallel process, this won't catch all files, as this only execute collection of files that on current job per process, eg; `16 files` per configured parallel job size

https://github.com/rectorphp/rector-src/blob/ac3a9412706652b2d93d900abc1dfef06be3758c/src/Config/RectorConfig.php#L100

Reply:
I am thinking of cache the list of files on very first execution, that's the possible way it can get all files on parallel process

```php
    public function run(Configuration $configuration, InputInterface $input): ProcessResult
    {
        $filePaths = $this->filesFinder->findFilesInPaths($configuration->getPaths(), $configuration);

        // no files found
        if ($filePaths === []) {
            return new ProcessResult([], []);
        }

        // cache here ...
        $this->cache->save(key, variableKey, $filePaths);
```

There is `Rector\Caching\Cache` service that may can be used for it.

then you pull from cache when exists on `processFiles()`

Reply:
@marcelthole I created example of cache usage on PR:

- https://github.com/rectorphp/rector-src/pull/6894

feel free to cherry-pick/copy :)